test/suites/optional/cert-manager.robot

```diff
@@ -16,100 +17,70 @@ Test Tags           cert-manager    certificates    tls
-...                         \ \ \ \ \ \ name: ${SECRET_NAME}
-...                         \ \ \ \ termination: edge
-...                         \ \ to:
-...                         \ \ \ \ kind: Service
-...                         \ \ \ \ name: hello-microshift
+${CERT_NAME}                    test-certificate
+${SECRET_NAME}                  test-cert-secret
+${ISSUER_NAME}                  test-issuer
+${CERT_COMMON_NAME}             example.com
+${CERT_DNS_NAME}                example.com
+${ROUTE_NAME}                   hello-app
+${CERT_ISSUER_YAML}             SEPARATOR=\n
+...                             ---
+...                             apiVersion: cert-manager.io/v1
+...                             kind: ClusterIssuer
+...                             metadata:
+...                             \ \ name: ${ISSUER_NAME}
+...                             spec:
+...                             \ \ selfSigned: {}
+
+${HTTP01_ISSUER_NAME}           letsencrypt-http01
+${HTTP01_CERT_NAME}             cert-from-${HTTP01_ISSUER_NAME}
+${HTTP01_SECRET_NAME}           ${HTTP01_CERT_NAME}
+${PEBBLE_DEPLOYMENT_FILE}       ./assets/cert-manager/pebble-server.yaml
 
 
 *** Test Cases ***
 Create Ingress route with Custom certificate
     [Documentation]    Create route with a custom certificate
     [Setup]    Run Keywords
-    ...    Setup Namespace
-    Apply YAML Manifest    ${CERT_ISSUER_YAML}
-    Oc Wait    -n ${CERTS_NAMESPACE} clusterissuer ${ISSUER_NAME}    --for="condition=Ready" --timeout=120s
-    Apply YAML Manifest    ${CERT_YAML}
-    Oc Wait    -n ${CERTS_NAMESPACE} certificate ${CERT_NAME}    --for="condition=Ready" --timeout=60s
-    Apply YAML Manifest    ${INGRESS_RBAC_YAML}
+    ${cert_issuer_yaml}=    Create Cert Issuer YAML
+    Apply YAML Manifest    ${cert_issuer_yaml}
+    Oc Wait    -n ${NAMESPACE} clusterissuer ${ISSUER_NAME}    --for="condition=Ready" --timeout=120s
+    ${cert_yaml}=    Create Certificate YAML For Test
+    Apply YAML Manifest    ${cert_yaml}
+    Oc Wait    -n ${NAMESPACE} certificate ${CERT_NAME}    --for="condition=Ready" --timeout=60s
+    ${rbac_yaml}=    Create Ingress RBAC YAML
+    Apply YAML Manifest    ${rbac_yaml}
     Deploy Hello MicroShift
-    Apply YAML Manifest    ${INGRESS_ROUTE_YAML}
-    Oc Wait    -n ${CERTS_NAMESPACE} route ${ROUTE_NAME}    --for=jsonpath='.status.ingress' --timeout=60s
+    ${route_yaml}=    Create Ingress Route YAML
+    Apply YAML Manifest    ${route_yaml}
+    Oc Wait    -n ${NAMESPACE} route ${ROUTE_NAME}    --for=jsonpath='.status.ingress' --timeout=60s
     [Teardown]    Run Keywords
     ...    Remove ClusterIssuer
 
+Test Cert manager with local acme server
+    [Documentation]    Test cert-manager with local ACME server (Pebble) using HTTP01 challenge
+    [Tags]    http01    acme
+    [Setup]    Setup Pebble Server    ${NAMESPACE}
+
+    ${dns_name}=    Generate Random HostName
+    Add Entry To Hosts    ${USHIFT_HOST}    ${dns_name}
```

Comment:
I dont think we need this keyword `Add Entry To Hosts`  since we are using DNS..

Reply:
@eslutsky I do not understand, where are we using DNS ? we are not using DNS anywhere.